types/redux-saga-tester/index.d.ts

```diff
@@ -45,7 +47,7 @@ export default class SagaTester<StateType extends object> {
      * Returns a promise that will resolve if the specified action is dispatched to the store.
      * @param futureOnly Causes waitFor to only resolve if the action is called in the future.
      */
-    waitFor(actionType: string, futureOnly?: boolean): PromiseLike<void>;
+    waitFor<T>(actionType: string, futureOnly?: boolean): PromiseLike<T>; // tslint:disable-line no-unnecessary-generics
```

Comment:
We could improve this a little by making the `actionType` constrained to the same as `action.type`
```suggestion
    waitFor<K extends string, T extends { type: K }>(actionType: K, futureOnly?: boolean): PromiseLike<T>; // tslint:disable-line no-unnecessary-generics
```